src/odemis/acq/test/path_test.py

```diff
@@ -978,21 +1002,246 @@ def test_set_path(self):
+        # Assert that actuator was moved according to mode given
+        self.assert_pos_as_in_mode(self.lenswitch, "spectral")
+        # No slit/spectrograph as they are not affecting the detector
+        self.assertAlmostEqual(self.spec_sel.position.value["x"], 0.026112848)
+
+        # Change positions back
+        self.optmngr.setPath("chamber-view").result()
+
+        self.optmngr.setPath(sps).result()
+        # Assert that actuator was moved according to mode given
+        self.assert_pos_as_in_mode(self.lenswitch, "spectral")
+        self.assertAlmostEqual(self.spec_sel.position.value["x"], 0.026112848)
+
+
+class Sparc2FourSpecPathTestCase(unittest.TestCase):
+    """
+    Tests to be run with a (simulated) SPARC2 with 2 spectrometers on the 
+    "integrated" spectrograph and 2 spectrometers on a spectrograph connected
+    via an optical fiber. In addition to check the handling of multiple external
+    spectrometers, it also tests the detectors with numbered roles (eg, spectrometer2) 
+    """
+    backend_was_running = False
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            test.start_backend(SPARC2_4SPEC_CONFIG)
+        except LookupError:
+            logging.info("A running backend is already found, skipping tests")
+            cls.backend_was_running = True
+            return
+        except IOError as exp:
+            logging.error(str(exp))
+            raise
+
+        # Microscope component
+        cls.microscope = model.getComponent(role="sparc2")
+        # Find CCD & SEM components
+        cls.ccd = model.getComponent(role="ccd0")
+        cls.ispec1 = model.getComponent(role="spectrometer0")  # wrapper of CCD
+        cls.ispec2 = model.getComponent(role="spectrometer1")
+        cls.espec1 = model.getComponent(role="spectrometer2")
+        cls.espec2 = model.getComponent(role="spectrometer3")
+        cls.specgraph = model.getComponent(role="spectrograph")
+        cls.spec_det_sel = model.getComponent(role="spec-det-selector")
+        cls.specgraph_dedicated = model.getComponent(role="spectrograph-dedicated")
+        cls.spec_dd_sel = model.getComponent(role="spec-ded-det-selector")
+        cls.ebeam = model.getComponent(role="e-beam")
+        cls.sed = model.getComponent(role="se-detector")
+        cls.lensmover = model.getComponent(role="lens-mover")
+        cls.lenswitch = model.getComponent(role="lens-switch")
+        cls.spec_sel = model.getComponent(role="spec-selector")
+        cls.slit = model.getComponent(role="slit-in-big")
+        cls.optmngr = path.OpticalPathManager(cls.microscope)
+
+    @classmethod
+    def tearDownClass(cls):
+        if cls.backend_was_running:
+            return
+        del cls.optmngr  # To garbage collect it
+        test.stop_backend()
+
+    def setUp(self):
+        if self.backend_was_running:
+            self.skipTest("Running backend found")
+
+    def assert_pos_as_in_mode(self, comp, mode):
```

Comment:
Is this an exact copy from the method on line 896? Wouldn't it be better then to define it somewhere separate of the test classes?

Reply:
Yeah... we have 3 or 4 copies of it. As it needs `self`, we would need to make a parent abstract class. As it's in the test cases, I feel it's not the worth the extra work.